sql/core/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveMetricView.scala

```diff
@@ -0,0 +1,348 @@
+ *   SELECT region_upper FROM sales_metrics GROUP BY 1
+ *   => SELECT upper(region) FROM sales_table WHERE product = 'product_1' GROUP BY 1
+ *
+ *   -- Measure only (aggregates entire dataset)
+ *   SELECT MEASURE(total_sales) FROM sales_metrics
+ *   => SELECT sum(amount) FROM sales_table WHERE product = 'product_1'
+ *
+ *   -- Dimensions + Measures (group by dimensions)
+ *   SELECT region, MEASURE(total_sales) FROM sales_metrics GROUP BY region
+ *   => SELECT region, sum(amount) FROM sales_table
+ *      WHERE product = 'product_1' GROUP BY region
+ * }}}
+ *
+ * The rule operates on unresolved plans and transforms [[MetricViewPlaceholder]] nodes
+ * into resolved logical plans that can be further optimized and executed.
+ */
+case class ResolveMetricView(session: SparkSession) extends Rule[LogicalPlan] {
+  private def parser: ParserInterface = session.sessionState.sqlParser
+  override def apply(plan: LogicalPlan): LogicalPlan = {
+    if (!plan.containsPattern(METRIC_VIEW_PLACEHOLDER)) {
+      return plan
+    }
+    plan.resolveOperatorsUp {
+      // CREATE PATH: to create a metric view, we need to analyze the metric view
+      // definition and get the output schema (with column metadata). Since the measures
+      // are aggregate functions, we need to use an Aggregate node and group by all
+      // dimensions to get the output schema.
+      case mvp: MetricViewPlaceholder if mvp.isCreate && mvp.child.resolved =>
+        val (dimensions, measures) = buildMetricViewOutput(mvp.desc)
+        Aggregate(
+          // group by all dimensions
+          dimensions.map(_.toAttribute).toSeq,
+          // select all dimensions and measures to get the final output (mostly data types)
+          (dimensions ++ measures).toSeq,
+          mvp.child
+        )
+
+      // SELECT PATH: to read a metric view, user will use the `MEASURE` aggregate function
+      // to read the measures, so it'll lead to an Aggregate node. This way, we only need to
+      // Resolve the Aggregate node based on the metric view output and then replace
+      // the AttributeReference of the metric view output to the actual expressions.
+      case node @ MetricViewReadOperation(metricView) =>
+        // step 1: parse the metric view definition
+        val (dimensions, measures) =
+          parseMetricViewColumns(metricView.outputMetrics, metricView.desc.select)
+
+        // step 2: build the Project node containing the dimensions
+        val dimensionExprs = dimensions.map(_.namedExpr)
+        // Drop the source columns if it conflicts with dimensions
+        val sourceOutput = metricView.child.output
+        // 1. hide the column conflict with dimensions
+        // 2. add an alias to the source column so they are stable with DeduplicateRelation
+        // 3. metric view output should use the same exprId
+        val sourceProjList = sourceOutput.filterNot { attr =>
+          // conflict with dimensions
+          metricView.outputMetrics
```

Comment:
`outputMetrics` contains both dimension and measure columns, shall we filter out dimension columns first before we look up the column name?